tasks/delete_local_filebucket.rb

```diff
@@ -0,0 +1,55 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+# frozen_string_literal: true
+
+require 'json'
+require 'puppet'
+
+params = JSON.parse(STDIN.read)
+force = params['force']
+require_relative File.join(params['_installdir'], 'puppet_agent', 'files', 'rb_task_helper.rb')
+
+module PuppetAgent
+  class DeleteLocalFilebucket
+    include PuppetAgent::RbTaskHelper
+
+    def initialize(force)
+      @force = force
+    end
+
+    def run
+      return error_result(
+        'puppet_agent/no-puppet-bin-error',
+        "Puppet executable '#{puppet_bin}' does not exist",
+      ) unless puppet_bin_present?
+
+      begin
+        FileUtils.remove_dir(clientbucketdir, force)
+        { "success": true }
+      rescue StandardError => e
+        error_result(
+          'puppet_agent/cannot-remove',
+          "#{e.class}: #{e.message}"
+        )
+      end
+    end
+
+    private
+
+    def clientbucketdir
+      options = {
+        failonfail: false,
+        override_locale: false,
+      }
+
+      command = "#{puppet_bin} config print clientbucketdir"
```

Comment:
@joshcooper thoughts on this? Could this not return the correct path if `clientbucketdir` is set in a different section?

Reply:
`clientbucketdir` can also be found without shelling out to puppet config print.
```
Puppet.initialize_settings
Puppet[:clientbucketdir]
```

Reply:
Looks like `clientbucketdir` is defined under the `agent` section in [defaults.rb](https://github.com/puppetlabs/puppet/blob/main/lib/puppet/defaults.rb#L1734). However, the puppet filebucket application doesn't enforce a specific run_mode, so it will ignore the value of `clientbucketdir` if set under `[agent]`.

Using this logic, I think we can rely on the `initialize_settings` approach as it would be a bit faster.

Reply:
One downside with `initialize_settings` is it can only be called once:

```shell
irb(main):001:0> require 'puppet'
=> true
irb(main):002:0> Puppet.initialize_settings
=> {}
irb(main):003:0> Puppet.initialize_settings
Traceback (most recent call last):
       16: from /home/josh/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.2.16/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
       15: from /home/josh/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.2.16/lib/bundler/cli.rb:30:in `dispatch'
       14: from /home/josh/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.2.16/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
       13: from /home/josh/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.2.16/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
       12: from /home/josh/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.2.16/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
       11: from /home/josh/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.2.16/lib/bundler/cli.rb:494:in `exec'
       10: from /home/josh/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.2.16/lib/bundler/cli/exec.rb:28:in `run'
        9: from /home/josh/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.2.16/lib/bundler/cli/exec.rb:63:in `kernel_load'
        8: from /home/josh/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/bundler-2.2.16/lib/bundler/cli/exec.rb:63:in `load'
        7: from /home/josh/.rbenv/versions/2.7.3/bin/irb:23:in `<top (required)>'
        6: from /home/josh/.rbenv/versions/2.7.3/bin/irb:23:in `load'
        5: from /home/josh/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        4: from (irb):3
        3: from /home/josh/work/puppet/lib/puppet.rb:153:in `initialize_settings'
        2: from /home/josh/work/puppet/lib/puppet.rb:178:in `do_initialize_settings_for_run_mode'
        1: from /home/josh/work/puppet/lib/puppet/settings.rb:305:in `initialize_global_settings'
Puppet::DevError (Attempting to initialize global default settings more than once!)
```

So it seems like there could be issues if this task was called twice in a plan (seems unlikely) or some other task also called `initialize_settings`?

I'd be inclined to just shell out.